store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/Converter.kt

```diff
@@ -0,0 +1,47 @@
+package org.mobilenativefoundation.store.store5
+
+interface Converter<Network : Any, Common : Any, SOT : Any> {
+    fun fromNetworkToCommon(network: Network): Common?
+    fun fromCommonToSOT(common: Common): SOT?
+    fun fromSOTToCommon(sourceOfTruth: SOT): Common?
+
+    class Builder<Network : Any, Common : Any, SOT : Any> {
+
+        private var fromCommonToSOT: ((input: Common) -> SOT)? = null
+        private var fromNetworkToCommon: ((input: Network) -> Common)? = null
```

Comment:
it looks like we can have a version of the converter that has:

fromNetworkToSOT
fromSOTToCommon
fromCommonToSOT

This would skip the need to make a converter going from network to common. Raises the question of whether that should be default as well. Why make someone pass a converter from network to common when they can give you a converter from network right to SOT

Reply:
Ooh yes great idea 👍🏽

Reply:
Will do separately in #504